experiments/storm/src/main/scala/io/gearpump/experiments/storm/producer/StormSpoutOutputCollector.scala

```diff
@@ -20,15 +20,20 @@ package io.gearpump.experiments.storm.producer
 
 import java.util.{List => JList}
 
-import backtype.storm.spout.ISpoutOutputCollector
+import backtype.storm.spout.{ISpout, ISpoutOutputCollector}
+import io.gearpump.TimeStamp
 import io.gearpump.experiments.storm.util.StormOutputCollector
 
 /**
  * this is used by Storm Spout to emit messages
  */
-private[storm] class StormSpoutOutputCollector(collector: StormOutputCollector) extends ISpoutOutputCollector {
+private[storm] class StormSpoutOutputCollector(collector: StormOutputCollector, spout: ISpout) extends ISpoutOutputCollector {
+  private var pendingMessages = Vector.empty[(TimeStamp, Object)]
```

Comment:
It also will OO quickly if you do nothing about ack or fail.
Check https://github.com/nathanmarz/storm-contrib/blob/master/storm-kafka/src/jvm/storm/kafka/PartitionManager.java#L38

There is a TreeSet